virttest/utils_test/__init__.py

```diff
@@ -242,7 +242,14 @@ def get_time(session, time_command, time_filter_re, time_format):
                 raise exceptions.TestError(msg)
             if re.search('PM', guest_time):
                 hour = re.findall('\d+ (\d+):', guest_time)[0]
-                hour = str(int(hour) + 12)
+                if int(hour) < 12:
+                    hour = str(int(hour) + 12)
+                guest_time = re.sub('\s\d+:', " %s:" % hour,
+                                    guest_time)[:-3]
+            elif re.search('AM', guest_time):
+                hour = re.findall('\d+ (\d+):', guest_time)[0]
+                if int(hour) == 12:
+                    hour = str(int(hour) - 12)
                 guest_time = re.sub('\s\d+:', " %s:" % hour,
```

Comment:
L253 and L250 are duplicate with L247 and L244,  how about replace L243~L256 with: 

`hour = re.findall('\d+ (\d+):', guest_time)[0]           
 if re.search('PM', guest_time):
     if int(hour) < 12:
         hour = str(int(hour) + 12)
elif re.search('AM', guest_time):
    if int(hour) == 12:
        hour = str(int(hour) - 12)
guest_time = re.sub('\s\d+:', " %s:" % hour, guest_time)[:-3]`